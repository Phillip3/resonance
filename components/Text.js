/**
 * Text component that uses the correct font-family
 * and supports different font weights
 */

import React from 'react';
import { Text } from 'react-native';

export default props => {
  let fontWeight = 'Regular';

  if (props.light) {
    fontWeight = 'Light';
  } else if (props.bold) {
    fontWeight = 'Bold';
  }

  return (
    <Text
      {...props}
      allowFontScaling={false}
      style={[{ fontFamily: `NunitoSans-${fontWeight}` }, props.style]}
    >
      {props.children}
    </Text>
  );
};
