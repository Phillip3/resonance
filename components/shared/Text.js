/**
 * Text component that uses the correct font-family
 * and supports different font weights
 */

import React from 'react';
import { Text } from 'react-native';

export default props => {
  let fontSize = 16;

  let fontWeight = 'Regular';

  if (props.light) {
    fontWeight = 'Light';
  } else if (props.bold) {
    fontWeight = 'Bold';
  }

  if (props.XL) {
    fontSize = 30;
  } else if (props.L) {
    fontSize = 20;
  } else if (props.M) {
    fontSize = 18;
  } else if (props.S) {
    fontSize = 14;
  } else if (props.XS) {
    fontSize = 12;
  }

  return (
    <Text
      allowFontScaling={false}
      selectable
      {...props}
      style={[
        { fontSize, fontFamily: `NunitoSans-${fontWeight}` },
        props.style,
      ]}
    >
      {props.children}
    </Text>
  );
};
