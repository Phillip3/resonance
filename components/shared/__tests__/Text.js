import React from 'react';
import { render } from 'react-native-testing-library';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import Text from '../Text';

describe('<Text/>', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  test('render text', async () => {
    const { toJSON } = render(<Text>ðŸ”¥</Text>);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render text with a light font', async () => {
    const { toJSON } = render(<Text light>ðŸ”¥</Text>);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render text with a bold font', async () => {
    const { toJSON } = render(<Text bold>ðŸ”¥</Text>);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render text with style', async () => {
    const { toJSON } = render(<Text style={{ fontSize: 30 }}>ðŸ”¥</Text>);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render text with extra props', async () => {
    const { toJSON } = render(
      <Text testID="text-with-some-props" disabled>
        ðŸ”¥
      </Text>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
