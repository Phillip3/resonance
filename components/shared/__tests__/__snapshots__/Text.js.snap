// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> render text 1`] = `
<Text
  allowFontScaling={false}
  selectable={true}
  style={
    Array [
      Object {
        "fontFamily": "NunitoSans-Regular",
        "fontSize": 16,
      },
      undefined,
    ]
  }
>
  ðŸ”¥
</Text>
`;

exports[`<Text/> render text with a bold font 1`] = `
<Text
  allowFontScaling={false}
  bold={true}
  selectable={true}
  style={
    Array [
      Object {
        "fontFamily": "NunitoSans-Bold",
        "fontSize": 16,
      },
      undefined,
    ]
  }
>
  ðŸ”¥
</Text>
`;

exports[`<Text/> render text with a light font 1`] = `
<Text
  allowFontScaling={false}
  light={true}
  selectable={true}
  style={
    Array [
      Object {
        "fontFamily": "NunitoSans-Light",
        "fontSize": 16,
      },
      undefined,
    ]
  }
>
  ðŸ”¥
</Text>
`;

exports[`<Text/> render text with extra props 1`] = `
<Text
  allowFontScaling={false}
  disabled={true}
  selectable={true}
  style={
    Array [
      Object {
        "fontFamily": "NunitoSans-Regular",
        "fontSize": 16,
      },
      undefined,
    ]
  }
  testID="text-with-some-props"
>
  ðŸ”¥
</Text>
`;

exports[`<Text/> render text with style 1`] = `
<Text
  allowFontScaling={false}
  selectable={true}
  style={
    Array [
      Object {
        "fontFamily": "NunitoSans-Regular",
        "fontSize": 16,
      },
      Object {
        "fontSize": 30,
      },
    ]
  }
>
  ðŸ”¥
</Text>
`;
